dot/core/service.go

```diff
@@ -127,40 +127,64 @@ func (s *Service) StorageRoot() (common.Hash, error) {
 
 // HandleBlockImport handles a block that was imported via the network
 func (s *Service) HandleBlockImport(block *types.Block, state *rtstorage.TrieState) error {
-	return s.handleBlock(block, state)
+	err := s.handleBlock(block, state)
+	if err != nil {
+		return fmt.Errorf("handling block: %s", err)
+	}
+
+	bestBlockHash := s.blockState.BestBlockHash()
+	isBestBlock := bestBlockHash.Equal(block.Header.Hash())
+
+	blockAnnounce, err := createBlockAnnounce(block, isBestBlock)
+	if err != nil {
+		return fmt.Errorf("creating block announce: %w", err)
+	}
+
+	s.net.GossipMessage(blockAnnounce)
```

Comment:
Is there a risk/is it allowed to gossip a block announce back to the peer that sent it to us?

Reply:
I am not sure about the risks/penalties (I did not see any while looking into substrate logs), I would say that the intended behavior is to don't send the message back but right now this is out of the scope of this PR, should we create an issue to address the creation of a method `GossipMessageExcluding(message, peers...)`? @timwu20 @danforbes

Reply:
I would need a little more information to be able to definitively answer this question or to be able to comment on the priority of such a task.